docs/concepts/advanced/extensions.md

```diff
@@ -169,6 +169,14 @@ class MyExtension(ComponentExtension):
         ctx.component_cls.my_attr = "my_value"
 ```
 
+!!! warning
+
+    The `name` attribute MUST be unique across all extensions.
```

Comment:
Unrelated to the linked issue, but added validation to extensions so that if there's duplicate name, or the name conflicts with any attributes or methods on the Component class, then we raise error.